docker/distribution_error.go

```diff
@@ -30,14 +30,18 @@ import (
 // errcode.Errors slice.
 var errNoErrorsInBody = errors.New("no error details found in HTTP response body")
 
-// unexpectedHTTPStatusError is returned when an unexpected HTTP status is
+// UnexpectedHTTPStatusError is returned when an unexpected HTTP status is
 // returned when making a registry api call.
-type unexpectedHTTPStatusError struct {
-	Status string
+type UnexpectedHTTPStatusError struct {
+	// StatusCode code as returned from the server, public so callers can
+	// match the exact code to make certain decisions if needed.
```

Comment:
(I’m not sure the “public …” part is relevant to external consumers, but it doesn’t hurt.)